packages/ibm-products/src/components/UserAvatar/UserAvatar.js

```diff
@@ -70,19 +72,40 @@ export let UserAvatar = React.forwardRef(
     const carbonPrefix = usePrefix();
     const icons = {
       user: {
-        md: <User size={32} />,
+        sm: (props) => <User size={16} {...props} />,
+        md: (props) => <User size={20} {...props} />,
+        lg: (props) => <User size={24} {...props} />,
+        xl: (props) => <User size={32} {...props} />,
       },
       group: {
-        md: <Group size={32} />,
+        sm: (props) => <Group size={16} {...props} />,
+        md: (props) => <Group size={20} {...props} />,
+        lg: (props) => <Group size={24} {...props} />,
+        xl: (props) => <Group size={32} {...props} />,
       },
     };
+    const formatInitials = () => {
+      if (name.length === 2) {
+        return name;
+      }
+      // RegEx takes in the display name and returns the first and last name. Thomas Watson and Thomas J. Watson
+      // both return JW.
+      return name
+        .match(/(^\S\S?|\b\S)?/g)
+        .join('')
+        .match(/(^\S|\S$)?/g)
+        .join('')
+        .toUpperCase();
+    };
     const getItem = (renderIcon) => {
       if (renderIcon === User) {
-        return icons.user['md'];
+        return icons.user[size];
       } else if (renderIcon === Group) {
-        return icons.group['md'];
+        return icons.group[size];
+      } else if (name) {
+        return formatInitials;
       } else {
-        return renderIcon;
+        return icons['user'][size];
```

Comment:
The corresponding icon size should always be the same based on the avatar size.
Rather than checking for `renderIcon`, I think it makes more sense to just render the icon[^1] that gets passed in using the sizes mapped below.

| Size  | Icon scale |
|------|-----------|
| `xl`   | `32`          |
| `lg`   | `24`          |
| `md` | `20`          |
| `sm` | `16`          |

[^1]: I think we have a few examples in `ButtonMenu`, `ComboButton`, `EditInPlace`/`InlineEdit` and a few other places like if you’d like to see some examples although they don’t include size.

Reply:
Changed based on @davidmenendez suggestion